File: C++/Medium/WrongAnswer/minimum-fuel-cost-to-report-to-the-capital_1.cpp
class Solution {
public:
    long long minimumFuelCost(vector<vector<int>>& roads, int seats) {
        
        int n = roads.size() + 1;
        vector<vector<int>> g(n);
        for (auto& r : roads) {
            g[r[0]].push_back(r[1]);
            g[r[1]].push_back(r[0]);
        }
        vector<int> sz(n);
        function<void(int, int)> dfs = [&](int u, int p) {
            sz[u] = 1;
            for (int v : g[u]) {
                if (v == p) continue;
                dfs(v, u);
                sz[u] += sz[v];
            }
        };
        dfs(0, -1);
        int m = seats;
        vector<vector<long long>> f(n, vector<long long>(m + 1));
        function<void(int, int)> solve = [&](int u, int p) {
            for (int v : g[u]) {
                if (v == p) continue;
                solve(v, u);
                for (int i = m; i >= 0; i--) {
                    for (int j = 0; j <= i; j++) {
                        f[u][i] = min(f[u][i], f[u][i - j] + f[v][j]);
                    }
                }
            }
            f[u][1] = sz[u] - 1;
        };
        solve(0, -1);
        long long ans = 1e18;
        for (int i = 0; i <= m; i++) ans = min(ans, f[0][i]);
        return ans;
    }
};

File: Java/Medium/WrongAnswer/minimum-fuel-cost-to-report-to-the-capital_1.java
class Solution {
    public long minimumFuelCost(int[][] roads, int seats) {
        
        int n = roads.length + 1;
        List<List<Integer>> graph = new ArrayList<>();
        for (int i = 0; i < n; i++) {
            graph.add(new ArrayList<>());
        }
        for (int[] road : roads) {
            graph.get(road[0]).add(road[1]);
            graph.get(road[1]).add(road[0]);
        }
        long[] dp = new long[n];
        long[] dp1 = new long[n];
        dfs(graph, dp, dp1, 0, -1, seats);
        return dp[0];
    }
    private void dfs(List<List<Integer>> graph, long[] dp, long[] dp1, int cur, int pre, int seats) {
        dp[cur] = 0;
        dp1[cur] = 0;
        List<Integer> children = graph.get(cur);
        for (int child : children) {
            if (child == pre) {
                continue;
            }
            dfs(graph, dp, dp1, child, cur, seats);
            dp[cur] += dp[child];
            dp1[cur] += dp[child];
        }
        Collections.sort(children, (a, b) -> Long.compare(dp1[b], dp1[a]));
        int cnt = 0;
        for (int child : children) {
            if (child == pre) {
                continue;
            }
            cnt += 1;
            dp[cur] += dp1[child];
            dp1[cur] += dp1[child];
            if (cnt == seats - 1) {
                break;
            }
        }
    }
}